pykokkos/interface/execution_space.py

```diff
@@ -22,4 +22,38 @@ def is_host_execution_space(space: ExecutionSpace) -> bool:
     if space is ExecutionSpace.Default:
         space = km.get_default_space()
 
-    return space in {ExecutionSpace.OpenMP, ExecutionSpace.Pthreads, ExecutionSpace.Serial}
\ No newline at end of file
+    return space in {ExecutionSpace.OpenMP, ExecutionSpace.Threads, ExecutionSpace.Serial}
+
+class ExecutionSpaceInstance:
+    """
+    An instance of the execution space, corresponding to Cuda/HIP
```

Comment:
If this only corresponds to streams, should we call it that way?

Reply:
Technically users can create an instance for OpenMP (which doesn't have streams), although I don't think it does anything right now. Future execution spaces might have use for this and they might not use the terminology "Streams".

Reply:
Stream would at least reflect the current usage.  In the future, you could reconsider the name if needed.  `ExecutionSpaceInstace` sounds like an instance of a class, but then instance of an instance sounds like a meta level. This is something to be discussed in the next pykokkos meeting, but seems unnecessarily complex in my view.